### ISSUE TITLE
DEP: change hipscat-->hats

### ISSUE THREAD

**Issue by bsipocz** [nasa-fornax/fornax-images#17]
There has been a name change and thus new releases, we should update the dependencies in the image, too.

(The full switchover should happen once @troyraen gives the green light that all our hipscat holdings have been replaced by HATS ones)

**Comment by troyraen** [comment]
IRSA has not released any catalogs in hipscat format. The only hipscat catalog used in Fornax notebooks is Panstarrs (MAST holding) in the Light Curves notebook here: https://github.com/nasa-fornax/fornax-demo-notebooks/blob/main/light_curves/code_src/panstarrs_functions.py.

**Comment by zoghbi-a** [comment]
We didn't update this because the data in `s3://irsa-mast-tike-spitzer-data/data/ZTF/dr20` which is being used for benchmarking is still in hipscat format. Is this going to change.

We'll wait until that is updated and the notebooks are updated too.

**Comment by bsipocz** [comment]
Can we have both during the transitioning period?

**Comment by zoghbi-a** [comment]
I think Zak from MAST was using the ZTF data in irsa-mast-tike-spitzer-data.

We cannot have both. lsdb version 0.3 and 0.4 are incompatible. The former uses hipscat and the latter uses hats. Are you suggesting having them in separate conda environments?

**Comment by troyraen** [comment]
> I think Zak from MAST was using the ZTF data in irsa-mast-tike-spitzer-data.

Good to know he's still using that, thank you. I'm planning to migrate that bucket to a different account, so either the bucket name will change or there will be a period when the data is unavailable. I'll reach out to individuals and also post generally on Slack before I start that.

> lsdb version 0.3 and 0.4 are incompatible

Oh right, thanks. I was only thinking about hipscat and hats being separate libraries, but of course lsdb is the main library here. In that case, I think your original thinking makes sense to leave it as-is until there are actual HATS products available.

**Comment by bsipocz** [comment]
> We cannot have both. lsdb version 0.3 and 0.4 are incompatible. The former uses hipscat and the latter uses hats. Are you suggesting having them in separate conda environments?

Good point, my thinking stopped at the point of `hipscat` and `hats` and wasn't thinking about the incompatibility in lsdb. Full agree on staying put for now.

**Comment by troyraen** [comment]
We are now using HATS only and we can proceed with dropping dependency for HiPSCat libraries. I believe that the switch to HATS happened in version 0.4 of both `hats` and `lsdb`. But, due to more HATS format changes, we will now require version >=0.5 of both libraries. There will likely be knock-on effects related to dependencies because lsdb v0.5 also now requires numpy >=2.0. This is being discussed in https://github.com/nasa-fornax/fornax-demo-notebooks/issues/395.

**Comment by zoghbi-a** [comment]
numpy<2 is still needed by `lightkurve`:
```
numpy 1.26.4 [required: >=1.26.4,<2.0a0] 
  -> oktopus 0.1.2 [required: >=0.1.2] 
    -> lightkurve==2.5.0
```
So, we'll need to keep it <2.0 for now. It will change in the next update we handle the conflicts in a consistent way.

**Comment by bsipocz** [comment]
Hmm, that's a bummer. We don't really use lightkurve for much but accessing tess data, if I recall correctly, and then we don't really use those for anything later on, so disabling tess for the sake of hats/lsdb compatibility

**Comment by bsipocz** [comment]
lightkurve is now numpy 2.0 compatible 🎉

**Comment by zoghbi-a** [comment]
Thanks @bsipocz,
I have the image that handles the conflicts setup with separate environment. If you update the notebook requirement files, it will pick them automatically.
I will just need to update the deployment to use it, and it will happen in a day or two.

**Comment by bsipocz** [comment]
The issue with this was that we require lightkurve and hats in the same notebook.
(I don't think it's appropriate to add the version pin in the requirement file as we don't really require the new version for the functionality, it's just for cross compatible reasons and the dependency resolver should already pick up on that, so it all should be fixed with a job retrigger I think)
